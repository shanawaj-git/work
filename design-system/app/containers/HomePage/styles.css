.swiper {
  max-width: 400px;
}
