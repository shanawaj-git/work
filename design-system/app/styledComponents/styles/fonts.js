const FONTS = {
  displayXXL: {
    fontFamily: 'Poppins',
    fontSize: '64px',
    lineHeight: '150%',
  },
  displayXL: {
    fontFamily: 'Poppins',
    fontSize: '48px',
    lineHeight: '150%',
  },
  displayL: {
    fontFamily: 'Poppins',
    fontSize: '36px',
    lineHeight: '150%',
  },
  displayM: {
    fontFamily: 'Poppins',
    fontSize: '26px',
    lineHeight: '150%',
  },
  bodyL: {
    fontFamily: 'Poppins',
    fontSize: '20px',
    lineHeight: '150%',
  },
  bodyM: {
    fontFamily: 'Poppins',
    fontSize: '16px',
    lineHeight: '150%',
  },
  bodyS: {
    fontFamily: 'Poppins',
    fontSize: '14px',
    lineHeight: '150%',
  },
  bodyXS: {
    fontFamily: 'Poppins',
    fontSize: '12px',
    lineHeight: '150%',
  },
};

export { FONTS };
