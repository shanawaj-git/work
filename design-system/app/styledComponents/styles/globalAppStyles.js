import { createGlobalStyle } from 'styled-components';

const GlobalAppStyles = createGlobalStyle`
@import url('https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap');

  html {
    box-sizing: border-box;
    text-size-adjust: 100%;
  }

  body {
    font-family: 'Poppins';
    font-weight: 400;
    -webkit-font-smoothing: antialiased;
    margin: 0;
    font-size: 1rem;
  }

  html * {
    box-sizing: inherit;
  }

  /* https://cssreset.com/scripts/eric-meyer-reset-css/ */
  /* http://html5doctor.com/html-5-reset-stylesheet/ */
  *, h1 {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
  }
  
  hr {
    border-top: 1px inset;
  }

  /* From  defaultRootStyle in /imports/app/molecules/Buttons/Button/styles */
  button {
    cursor: pointer;

    
  }

  body.using-mouse &:focus {
    outline: none;
    box-shadow: none;
  }

  li.root {
    display: block;
  }

  /* This is a rule to hide all marketing pixels which are attached to body. */
  /* In some cases they are causing visual artifacts, that's why we always want to be sure */
  /* that they are hidden */
  body > img {
    position: absolute;
    top: 0;
    right: 200%;
  }

  img {
    pointer-events: none;
  }
`;

export { GlobalAppStyles };
