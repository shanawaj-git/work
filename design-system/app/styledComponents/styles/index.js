export * from './globalAppStyles';
export * from './theme';
export * from './fonts';

export { default as defaultTheme } from './theme';
export { createMaterialUiTheme } from './createMuiTheme';
