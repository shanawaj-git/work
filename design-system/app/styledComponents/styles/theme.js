import { css } from 'styled-components';

import { FONTS } from './fonts';
import * as viewports from '../../constants/viewports';

const color = Object.freeze({
  p100: '#FAC2B3',
  p200: '#F6947A',
  p300: '#F26641',
  p400: '#CE5737',

  a100: '#FFE0BC',
  a200: '#FFC889',
  a300: '#F5AB56',
  a400: '#F99727',

  b100: '#C4F8CC',
  b200: '#93E8A1',
  b300: '#70D280',
  b400: '#44C058',

  c100: '#B9D9FD',
  c200: '#84B7F1',
  c300: '#2575CD',
  c400: '#5E9EE6',

  d100: '#AAE4FB',
  d200: '#87D4F3',
  d300: '#5EC0E7',
  d400: '#3FA9D3',

  g100: '#F6F6F8',
  g200: '#DFE4ED',
  g300: '#D0D0D8',
  g400: '#7B7B8E',
  g500: '#13171C',
  g600: '#F2F2F7',

  fontPrimary: '#3A3A3A',
  fontSecondary: '#9A9D9D',
  'font-tertiary': '#3C3C43CC',

  white: '#FFFFFF',
  black: '#000000',
  error1: '#FFE4E4',
  error2: '#FD9490',
  error3: '#FA605A',
  error4: '#ED2C25',
});

const typo = Object.freeze(
  Object.keys(FONTS).reduce((acc, key) => {
    acc[key] = css`
      font-size: ${FONTS[key].fontSize};
      font-family: ${FONTS[key].fontFamily};
      line-height: ${FONTS[key].lineHeight};
    `;

    return acc;
  }, {}),
);

const typoSelector = typoKey => props => props.theme.typo[typoKey];

const fontWeight = Object.freeze({
  book: css`
    font-weight: 400;
  `,
  medium: css`
    font-weight: 500;
  `,
  demi: css`
    font-weight: 600;
  `,
  bold: css`
    font-weight: 700;
  `,
});

const boxShadowFocus = css`
  &:focus {
    box-shadow: 0px 0px 10px ${color.d100};
  }
`;

const boxShadow = Object.freeze({
  level1: css`
    box-shadow: 0px 3px 10px rgba(0, 0, 0, 0.07);
  `,
  level2: css`
    box-shadow: 0px 6px 10px rgba(0, 0, 0, 0.07);
  `,
  level3: css`
    box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.07);
  `,
  level4: css`
    box-shadow: 0px 14px 28px rgba(0, 0, 0, 0.1);
  `,
  level5: css`
    box-shadow: 0px 19px 38px rgba(0, 0, 0, 0.15);
  `,
  focus: css`
    box-shadow: 0px 0px 10px ${color.d100};
  `,
});

const border = Object.freeze({
  level1: '0.3px solid rgba(130, 130, 130, 0.5)',
});

const layer = Object.freeze({
  hidden: -1,
  select: 5,
  newQuoteButton: 45,
  navbar: 50,
  tooltip: 60,
  mobileSocialFooter: 70,
  popup: 90,
  modal: 100,
  slideOut: 1101,
  welcomeModal: 1301,
});

// https://material-ui.com/customization/default-theme/?expend-path=$.breakpoints.values
const breakpoint = Object.freeze({
  // this prop is needed because webapp/imports/styledComponents/Grid/Column.js uses xs viewport
  xs: 0,
  sm: 0,
  md: viewports.tablet.width,
  lg: viewports.desktop.width,
});

// https://getbootstrap.com/docs/4.3/layout/overview/
const query = {
  min: {
    solve: width => `(min-width: ${width}px)`,
    tablet: `(min-width: ${breakpoint.md}px)`,
  },
  max: {
    solve: width => `(max-width: ${width - 0.02}px)`,
    tablet: `(max-width: ${breakpoint.lg - 0.02}px)`,
  },
  mobile: `(max-width: ${breakpoint.md - 0.02}px)`,
  tablet: `(min-width: ${breakpoint.md}px) and (max-width: ${breakpoint.lg -
    0.02}px)`,
  desktop: `(min-width: ${breakpoint.lg}px)`,
  dpr2x: `(min-resolution: 2dppx), (min-resolution: 192dpi)`,
  mobileDpr2x: `(max-width: ${breakpoint.md -
    0.02}px) and (min-resolution: 2dppx), (max-width: ${breakpoint.md -
    0.02}px) and (min-resolution: 192dpi)`,
};

Object.keys(breakpoint).forEach(key => {
  query.min[key] = query.min.solve(breakpoint[key]);
  query.max[key] = query.max.solve(breakpoint[key]);
});

Object.freeze(query);
Object.freeze(query.min);
Object.freeze(query.max);

const defaultTheme = Object.freeze({
  color,
  typo,
  border,
  boxShadow,
  boxShadowFocus,
  layer,
  breakpoint,
  query,
  fontWeight,
});

export {
  color,
  typo,
  border,
  boxShadow,
  boxShadowFocus,
  layer,
  breakpoint,
  query,
  fontWeight,
  typoSelector,
};

export default defaultTheme;
