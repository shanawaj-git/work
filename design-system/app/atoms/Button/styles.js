import { get } from 'lodash';
import { FONTS } from 'styledComponents/styles';

import { boxShadowSelector, colorSelector } from 'themes';

export const defaultRootStyle = {
  background: 'transparent',
  border: 'none',
  cursor: 'pointer',
  fontFamily: 'Poppins',
  fontWeight: 'demi',
};

export const defaultTextStyle = {
  padding: '8px 16px',
};

export const defaultLabelStyle = {
  textTransform: 'none',
  paddingLeft: '6px',
  paddingRight: '6px',
};

const primaryStyle = props => ({
  color: get(props, 'theme.overrides.button.color', colorSelector('white')),
  backgroundColor: get(
    props,
    'theme.overrides.button.backgroundColor',
    colorSelector('P300'),
  ),
  borderRadius: get(props, 'theme.overrides.button.borderRadius', '32px'),
  borderWidth: '0',
  textTransform: 'capitalize',
  boxShadow: 'none',
  hover: {
    backgroundColor: get(
      props,
      'theme.overrides.button.backgroundColor',
      colorSelector('P300'),
    ),
    color: get(props, 'theme.overrides.button.color', colorSelector('white')),
  },
  active: {
    color: get(props, 'theme.overrides.button.color', colorSelector('P300')),
    backgroundColor: colorSelector('P300'),
  },
  disabled: {
    color: colorSelector('C200'),
    backgroundColor: `${colorSelector('P300')}7F`,
  },
  loading: {
    backgroundColor: colorSelector('P400'),
  },
});

export const styles = {
  default: primaryStyle,
  primary: primaryStyle,
  secondary: props => ({
    color: get(props, 'theme.overrides.button.color', colorSelector('P300')),
    backgroundColor: colorSelector('transparent'),
    borderWidth: '2px',
    borderRadius: get(props, 'theme.overrides.button.borderRadius', '32px'),
    borderColor: colorSelector('P300'),
    borderStyle: 'solid',
    textTransform: 'capitalize',
    hover: {
      backgroundColor: colorSelector('P300'),
      boxShadow: boxShadowSelector('level2'),
      color: get(props, 'theme.overrides.button.color', colorSelector('white')),
    },
    active: {
      color: get(props, 'theme.overrides.button.color', colorSelector('white')),
      backgroundColor: colorSelector('P400'),
    },
    disabled: {
      color: colorSelector('C400'),
      backgroundColor: colorSelector('G200'),
      borderStyle: 'none',
    },
    loading: {
      color: get(props, 'theme.overrides.button.color', colorSelector('white')),
      backgroundColor: colorSelector('P400'),
    },
  }),
  ghost: {
    color: colorSelector('P300'),
    backgroundColor: 'transparent',
    borderWidth: 0,
    hover: {
      color: colorSelector('P400'),
      backgroundColor: 'transparent',
    },
    active: {
      color: colorSelector('P400'),
      backgroundColor: 'transparent',
    },
    disabled: {
      color: colorSelector('G400'),
      backgroundColor: 'transparent',
    },
    loading: {
      color: colorSelector('P400'),
      backgroundColor: 'transparent',
    },
  },
  ghostSecondary: {
    color: colorSelector('G400'),
    backgroundColor: 'transparent',
    borderWidth: 0,
    hover: {
      color: colorSelector('G400'),
      backgroundColor: 'transparent',
    },
    active: {
      color: colorSelector('G400'),
      backgroundColor: 'transparent',
    },
    disabled: {
      color: colorSelector('G400'),
      backgroundColor: 'transparent',
    },
    loading: {
      color: colorSelector('G400'),
      backgroundColor: 'transparent',
    },
  },
  alert: {
    color: colorSelector('white'),
    backgroundColor: colorSelector('G500'),
    hover: {
      boxShadow: boxShadowSelector('level2'),
      backgroundColor: colorSelector('G500'),
    },
    active: {
      backgroundColor: colorSelector('G500'),
    },
    disabled: {
      color: colorSelector('G400'),
      backgroundColor: colorSelector('G200'),
    },
    loading: {},
  },
};

export const sizeStyles = {
  default: () => ({
    height: '50px',
    width: '140px',
    fontSize: FONTS.bodyM.fontSize,
    lineHeight: FONTS.bodyM.lineHeight,
    padding: 0,
    textPadding: defaultTextStyle.padding,
    labelPaddingLeft: defaultLabelStyle.paddingLeft,
    labelPaddingRight: defaultLabelStyle.paddingRight,
  }),
  ghost: {
    height: 'fit-content',
    width: 'fit-content',
    fontSize: FONTS.bodyM.fontSize,
    lineHeight: FONTS.bodyM.lineHeight,
    borderRadius: '0',
    padding: '0',
    textPadding: '0',
    labelPaddingLeft: '0',
    labelPaddingRight: '0',
  },
  icon: {
    height: '50px',
    width: '50px',
    fontSize: FONTS.bodyS.fontSize,
    lineHeight: FONTS.bodyS.lineHeight,
    borderRadius: '25px',
    padding: 0,
    textPadding: '8px',
    labelPaddingLeft: '0',
    labelPaddingRight: '0',
  },
  small: () => ({
    height: '40px',
    width: '100px',
    fontSize: FONTS.bodyS.fontSize,
    lineHeight: FONTS.bodyS.lineHeight,
    padding: 0,
    textPadding: defaultTextStyle.padding,
    labelPaddingLeft: defaultLabelStyle.paddingLeft,
    labelPaddingRight: defaultLabelStyle.paddingRight,
  }),
  medium: () => ({
    height: '50px',
    width: '140px',
    fontSize: FONTS.bodyM.fontSize,
    lineHeight: FONTS.bodyM.lineHeight,
    padding: 0,
    textPadding: defaultTextStyle.padding,
    labelPaddingLeft: defaultLabelStyle.paddingLeft,
    labelPaddingRight: defaultLabelStyle.paddingRight,
  }),
  large: () => ({
    height: '60px',
    width: '200px',
    fontSize: FONTS.bodyL.fontSize,
    lineHeight: FONTS.bodyL.lineHeight,
    padding: 0,
    textPadding: defaultTextStyle.padding,
    labelPaddingLeft: defaultLabelStyle.paddingLeft,
    labelPaddingRight: defaultLabelStyle.paddingRight,
  }),
};

/**
 * Based on Material-ui CircularProgress attributes
 */
export const buttonLoadingStyles = {
  default: {
    inlineStyle: {
      position: 'inherit',
      color: '#FFFFFF',
    },
    size: 18,
    thickness: 7,
  },
  ghost: {
    inlineStyle: {
      position: 'inherit',
      color: colorSelector('P300'),
    },
    size: 18,
    thickness: 7,
  },
};
