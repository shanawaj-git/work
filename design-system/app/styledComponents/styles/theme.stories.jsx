import React from 'react';
import StoryWrapper from 'molecules/utility/StoryWrapper';
import { color } from './theme';
const ColorCard = props => {
  const { color: colors } = props;
  return (
    <div
      style={{
        display: 'flex',
        width: '800px',
        flexDirection: 'row',
        flexFlow: 'wrap',
      }}
    >
      {Object.keys(colors).map(colorKey => {
        return (
          <div>
            <div
              style={{
                backgroundColor: colors[colorKey],
                width: '200px',
                height: '100px',
              }}
            >
              <div
                style={{
                  background: 'white',
                  display: 'inline',
                  fontWeight: 'bold',
                  textTransform: 'uppercase',
                }}
              >
                {colorKey + ` - ` + colors[colorKey]}
              </div>
            </div>
            <br />
          </div>
        );
      })}
    </div>
  );
};
export default {
  title: 'Tokens/React/Colors',
  component: ColorCard,
  argTypes: {
    themeType: {
      control: { type: 'select', options: ['DEFAULT', 'MPC'] },
    },
    color: { table: { disable: true } },
  },
  decorators: [Story => <StoryWrapper>{Story()}</StoryWrapper>],
};

const Template = args => <ColorCard {...args} />;

export const Colors = Template.bind({});
Colors.args = {
  color,
};
