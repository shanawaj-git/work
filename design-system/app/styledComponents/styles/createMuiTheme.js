import { createTheme } from '@mui/material/styles';

const createMaterialUiTheme = (theme, options) => {
  const { breakpoint, color } = theme;
  const opts = options || {};

  return createTheme({
    palette: {
      primary: {
        light: color.p300,
        main: color.p300,
        dark: color.p300,
        contrastText: color.c200,
      },
      error: {
        light: color.error4,
        main: color.error4,
        dark: color.error4,
        contrastText: color.error4,
      },
      background: {
        default: color.g100,
      },
    },
    breakpoints: {
      values: {
        ...breakpoint,
      },
    },
    typography: {
      useNextVariants: true,
      fontFamily: ['Poppins'].join(','),
    },
    overrides: {
      MuiPickersToolbarText: {
        toolbarTxt: {
          color: color.white,
        },
        toolbarBtnSelected: {
          color: color.white,
        },
      },
      MuiPickersDay: {
        daySelected: {
          color: color.white,
        },
      },
      ...opts.overrides,
    },
  });
};

export { createMaterialUiTheme };
