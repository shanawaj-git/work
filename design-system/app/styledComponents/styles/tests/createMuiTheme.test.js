import * as dependency from '@mui/material/styles';

import { createMaterialUiTheme } from '../createMuiTheme';

jest.mock('@mui/material/styles', () => ({
  createTheme: jest
    .fn()
    .mockImplementation(() => ({ theme: { color: 'red' } })),
}));
describe('createMuiTheme', () => {
  const input = {
    breakpoint: [],
    color: 'red',
    overrides: [
      {
        background: 'yellow',
      },
    ],
  };
  it('creates theme successfully', () => {
    const getRequestMock = jest.spyOn(dependency, 'createTheme');

    dependency.createTheme = jest.fn();
    expect(createMaterialUiTheme(input)).toMatchSnapshot();
    expect(getRequestMock).toHaveBeenCalledTimes(1);
  });
});
