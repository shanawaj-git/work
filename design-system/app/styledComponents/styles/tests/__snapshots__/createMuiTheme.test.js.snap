// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMuiTheme creates theme successfully 1`] = `
Object {
  "theme": Object {
    "color": "red",
  },
}
`;
